examples/emulator_examples/__init__.py

```diff
@@ -0,0 +1,2 @@
+# !/usr/bin/python3
+# -*- coding: utf-8 -*-
```

Comment:
> 1. For blocking tests, the user should have a warning about it, or a way to terminate it. Also they don't play nicely with CI. I wonder if we should remove them and keep just their manual-override version.

Will remove the blocking tests.

> 2. For non-blocking tests that are copied from the state_machine_examples folder, I wonder if we can find a way to include the code from the other modules, rather than rewrite, and inject the emulator before the run. In such way, only one version of the code is maintained and they would never run out of sync. However using the current design, i.e, using Bpod(emulator_mode), would require a workaround

Sounds reasonable. Let me return to this one once I've cleaned up the examples.

> On a second thought, I think we should have at least one example in this PR where we check the resulting current_trial output, otherwise there is nothing here that tests that part.

Since these were intended to be high-level behavioral tests for the user and not unit tests, I think we should avoid unit-test-like checks (e.g. assert) for these. Unit tests should be created as part of another PR.

Reply:
> Since these were intended to be high-level behavioral tests for the user and not unit tests, I think we should avoid unit-test-like checks (e.g. assert) for these. Unit tests should be created as part of another PR.

True, however the core value of the emulator is to produce an output raw-events/raw-states as if it works in the real life. In such way, the user can test the update of trial settings according to the animal action (e.g as in `updateCustomFields()` in `Mouse2AFC`).
I'm not proposing that we make sweep test for the resulting raw-events/states in this PR, but at least one test should check that we get any sensible output.

Reply:
Sure, I will create one.